Polymer/polymer-cli
# Lazy load big dependencies, reducing file size to 118MB production install

This PR is a proposal for lazy-loading big dependencies of the CLI as a result of a discussion yesterday on Slack.

`npm install --production` before: 257MB
`npm install --production` after: 118MB

Total file size reduction roughly 55%.

## Requiring global packages

Removing the dependencies on `web-component-tester` and `yeoman-generator` requires loading the packages in the global npm package space. Natively `require(...)` does not do this, however the package [`requireg`](https://github.com/h2non/requireg) contains a resolver that does load the global package. It will first try to fetch locally and if that fails (e.g. not a dev environment) try to fetch globally in multiple possible places. The package is 580KB which is not too bad I think.

## User workflow

Initially installing the CLI will therefore not install the dependencies. However, when the user runs `polymer init` or `polymer test`, a helpful message is shown that the packages are required to be installed.

### How to test this branch

1. `rm -rf node_modules`
2. `npm install`
3. `npm run build`
4. `rm -rf node_modules`
5. `npm install --production`
6. `npm uninstall -g yeoman-generator yeoman-environment`
7. `node bin/polymer.js init`
8. Observe that the message "error:   cli runtime exception: Polymer init requires `yeoman-generator` and `yeoman-environment` to be installed globally. Install with `npm install -g i yeoman-generator yeoman-environment`" is shown in the console
9. `npm install -g yeoman-generator yeoman-environment`
10. `node bin/polymer.js init`
11. Observe that the normal generator options are shown in the console and no error


Fixes #364
